### ISSUE TITLE
Its not able to read Chinese characters, its displaying ? instaed of characters

### ISSUE THREAD

**Issue by SupriyaPorwal** [Sicos1977/MSGReader#58]
using (var outlookMsg = new MsgReader.Outlook.Storage.Message(pathPSgFile))
            {
                Console.WriteLine("Subject: {0}", outlookMsg.Subject);
                Console.WriteLine("Body: {0}", outlookMsg.BodyText);

                Console.WriteLine("{0} Recipients", outlookMsg.Recipients.Count);
                foreach (MsgReader.Outlook.Storage.Recipient recip in outlookMsg.Recipients)
                {
                    Console.WriteLine(" {0}:{1}", recip.Type, recip.Email);
                }

                Console.WriteLine("{0} Attachments", outlookMsg.Attachments.Count);
                foreach (MsgReader.Outlook.Storage.Attachment attach in outlookMsg.Attachments)
                {
                    Console.WriteLine(" {0}, {1}b", attach.FileName, attach.Data.Length);
                }

Attached is msg template

            }
[IDENTIFIER  HTML Test Mail w  Att - Smoke-Test.zip](https://github.com/Sicos1977/MSGReader/files/885347/IDENTIFIER.HTML.Test.Mail.w.Att.-.Smoke-Test.zip)

**Comment by SupriyaPorwal** [comment]
Hello,

In my Email template there is some special letters like chinese/Japanese.
I am not able to read these character.Its displaying ? instead of this characters.

Am I doing anything wrong here?

Thanks

**Comment by Sicos1977** [comment]
Hi,

I can reproduce the problem, I get the same question marks over here. I try to look into it this weekend.

Greetings,
Kees van Spelde

**Comment by erhanalankus** [comment]
Hi,
I'm passing an .msg file to MsgReader.Outlook.Storage.Message() that contains Italian text.
I'm having trouble with non-english characters such as:

à character is read into the Storage.Message.BodyHtml as Ã
’ becomes â€™
ù becomes Ã¹
è becomes Ã¨

Thank you for your work,
Erhan

**Comment by Sicos1977** [comment]
Can you send me the file that is giving the problem? Please send it to sicos2002@hotmail.com and zip the msg file before sending. If you don't zip it Hotmail will convert it to eml and makes the file useless for me.

Greetings,
Kees

**Comment by erhanalankus** [comment]
Hi Kees,
I sent the file.

Thanks a lot,
Erhan

**Comment by erhanalankus** [comment]
I solved my issue by adding utf-8 charset meta tag:

`using (var msg = new MsgReader.Outlook.Storage.Message(filePath))`
`var htmlBody = msg.BodyHtml;`
`htmlBody = htmlBody.Replace("</head>", "<meta charset=\"UTF-8\"/></head>");`

Chinese characters are still broken though.

**Comment by Sicos1977** [comment]
Good to hear that you solved it that way. I try to find a fix for the Chinese chars but at the moment I'm a little bit busy

**Comment by erhanalankus** [comment]
Here are two msg files. They contain the same characters. Three latin and three Chinese characters.
One msg file is converted to html succesfully, one fails. Converted html files are included.

I hope it helps.

https://www.dropbox.com/s/im6yd9oe9qfmgbp/0602-114221.zip?dl=0

**Comment by Shashidanand** [comment]
I am too facing similar issue where Chinese characters are not rendered correctly in html body.

**Comment by Sicos1977** [comment]
Can you send me the MSG file and some screenshots to show what is going wrong? Please send it to sicos2002@hotmail.com, zip the msg file before sending.

**Comment by odee2004** [comment]
still having this issue with Japanese characters. Tried adding the meta tag that was mentioned above but it did not work. Any other work around?

**Comment by Sicos1977** [comment]
I got your mail so I reopened this issue. I'll try to look into it this week.

**Comment by Sicos1977** [comment]
Can you get the latest source from GitHub and see if that fixes your problem?

**Comment by mvpkenlin** [comment]
@Sicos1977 Thanks for your quick fix. It is working on my sent sample. 

However, I tried another msg file but it turns out to be all ????????. I tried to copy the chinese word from the original msg and paste to a new email and send to myself again. Then save the new email into msg (in unicode and non-unicode). Your latest code could display it (the copied email). 

I would like to send you the "problem" msg that I have in my hand, but it contains some sensitive information. I am asking the sender to send me a new one with removing some sensitive information. so that I could send u that msg file if it still cannot display.

**Comment by Sicos1977** [comment]
This commit should fix the error about rtfpipe not being signed --> 473f18e95586d49a4416a13d6cbc826070432e49

**Comment by Sicos1977** [comment]
And this commit should fix the problem that shows you ???????? instead of the correct Chinese chars. 13093209d959908a4c75938fea574f05063f5cd4

Thanks for helping me out with examples since I can't read Chinese :-)

**Comment by mvpkenlin** [comment]
@Sicos1977 Tested and all of the sample msg that I have in my hand, are all displaying correctly. Thanks a lot.

**Comment by Sicos1977** [comment]
A new package has been published on NuGet --> https://www.nuget.org/packages/MSGReader/

**Comment by odee2004** [comment]
Just want to add a note here regarding this one. We used an older version of MsgReader without the .net standard depedency, I think it was 2.1.5 when we encountered that issue.

There's a workaround to be able to parse asian characters (tested Chinese, Korean and Japanese) by getting body of the message in RTF format Storage.Message.BodyRtf then used RTF Pipe to convert it to Html format (Rtf.ToHtml).
